cirq/circuits/circuit.py

```diff
@@ -1379,17 +1401,20 @@ def apply_unitary_effect_to_state(
 
         qs = ops.QubitOrder.as_qubit_order(qubit_order).order_for(
             self.all_qubits().union(qubits_that_should_be_present))
-        n = len(qs)
+
+        # Force qubits to have dimension at least 2 for backwards compatibility.
+        qid_shape = self.max_qid_shape(qs, qid_min=2)
```

Comment:
Do you have an example where this fails? I'm okay with this failing on user gates that didn't account for it, because the user will also be providing the shape that causes the failure.

Reply:
This happens when there is a qubit that has no gates operating on it (e.g. when explicitly specifying a list of qubits).  There are several test cases that fail when `qid_min=1`.